python/lsst/ts/salkafka/component_producer_set.py

```diff
@@ -22,34 +22,73 @@
 
 __all__ = ["ComponentProducerSet"]
 
-
 import argparse
 import asyncio
+import concurrent.futures
+import functools
 import logging
+import multiprocessing
 import os
 import pathlib
+import psutil
+import queue
 import signal
+import sys
+import traceback
 
 from lsst.ts import salobj
-from lsst.ts import salkafka
+from .topic_names_set import TopicNamesSet
+from .component_producer import ComponentProducer
+from .kafka_producer_factory import KafkaConfiguration, KafkaProducerFactory
+
+
+def asyncio_run_method(method, **kargs):
```

Comment:
Note to @tribeiro : this could be replaced by a pair of tested calls to `functools.partial`. I have mixed feeling. I think this tiny function makes the code a bit easier to read, but the user has to look up what it does. The other way looks something like this (where this function is called, around line 356). I have not tested this, but I think it will work:
```
loop.run_in_executor(
    pool,
    functools.partial(
        asyncio.run,
        functools.partial(
            self.create_producer_subprocess,
            kafka_config=self.kafka_config,
            ...
        ),
    ),
)
```
If you have a strong opinion let me know. I'll be happy to change it.

Reply:
Uhmm I don't think I understand the reason for using this method. Why not just use `asyncio.run` directly?